case-lib/lib.sh

```diff
@@ -927,14 +965,32 @@ arecord_opts()
 	# shellcheck disable=SC2086
         tinycap $SOF_ALSA_OPTS $SOF_ARECORD_OPTS "$file" -D "$card_nr" -d "$dev_nr" -c "$channel" -t "$duration" -r "$rate" -b "$format"
     elif [[ "$SOF_ALSA_TOOL" = "alsa" ]]; then
-        dlogc "arecord $SOF_ALSA_OPTS $SOF_ARECORD_OPTS $*"
-        # shellcheck disable=SC2086
-        arecord $SOF_ALSA_OPTS $SOF_ARECORD_OPTS "$@"
+        if [[ "$SOF_TEST_PIPEWIRE" == true ]]; then
+            dlogc "timeout -k 60 30 arecord $SOF_ALSA_OPTS $SOF_ARECORD_OPTS $*"  # option -d doesn't work with pipewire so we need timeout
+            # shellcheck disable=SC2086
+            timeout -k 60 30 arecord $SOF_ALSA_OPTS $SOF_ARECORD_OPTS "$@"
+        else
+            dlogc "arecord $SOF_ALSA_OPTS $SOF_ARECORD_OPTS $*"
+            # shellcheck disable=SC2086
+            arecord $SOF_ALSA_OPTS $SOF_ARECORD_OPTS "$@"
+        fi  
     else
         die "Unknown ALSA tool: ${SOF_ALSA_TOOL}"
     fi
 }
 
+# Get the ID for a given sink or source type, e.g. "Speaker" or "Headphones"
+get_id_of_pipewire_endpoint()
+{
+    # $ wpctl status returns list of all endpoints managed by wireplumber. We filter by given sink/source type, which returns something like this: 
+    # │  *   48. sof-soundwire Headphones            [vol: 0.40]  (or without the * when it's not the current default)
+    # We filter out everything but ID, and only take the first line of the output (if there's more that one object of that type we ignore the rest)
+
+    local object_name="$1"
+    object_id=$(wpctl status | grep -i "$object_name" | tr -d '*' | awk '{print $2}' | tr -d '.' | head -n 1)
```

Comment:
The output of `wpctl status --name` looks a bit more "deterministic" and machine parsable but I don't know pipewire.

Reply:
A single `awk` invocation can do all that (and more)
```suggestion
    object_id=$(wpctl status | awk "/$object_name/"'{ sub("*",""); sub("\\.","", $2); print $2; exit 0; }')
```

Reply:
You can even do this:

```shell
local object_id
object_id=$(wpctl status |
   awk "/$object_name/"' { sub("*",""); sub("\\.","", $2); print $2; found=1; exit 0 } 
         END { if (!found) exit 1 }') || {
  error $object_name not found ...
  ...
} 

```